airflowHPC/dags/gmx_trigger.py

```diff
@@ -0,0 +1,314 @@
+        output_dir_outer=output_dir,
+        output_dir_inner=f"lambda_{coul_lambda}",
+        counter=iteration_num,
+        num_simulations=num_simulations,
+    )
+
+    grompp = ResourceGmxOperatorDataclass.partial(
+        task_id="grompp",
+        executor_config={
+            "mpi_ranks": 1,
+            "cpus_per_task": 2,
+            "gpus": 0,
+            "gpu_type": None,
+        },
+        gmx_executable="gmx_mpi",
+    ).expand(input_data=grompp_input_list)
+    grompp_input_list >> grompp
+
+    mdrun_input_list = (
+        update_gmxapi_input.override(task_id="mdrun_input_list")
+        .partial(
+            args=["mdrun", "-v", "-deffnm", output_name, "-ntomp", "2"],
+            input_files_keys={"-s": "-o"},
+            output_files={
+                "-c": f"{output_name}.gro",
+                "-x": f"{output_name}.xtc",
+                "-dhdl": "dhdl.xvg",
+            },
+        )
+        .expand(gmxapi_output=grompp.output)
+    )
+    mdrun = ResourceGmxOperatorDataclass.partial(
+        task_id="mdrun",
+        executor_config={
+            "mpi_ranks": 1,
+            "cpus_per_task": 2,
+            "gpus": 0,
+            "gpu_type": None,
+        },
+        gmx_executable="gmx_mpi",
+    ).expand(input_data=mdrun_input_list)
+    grompp >> mdrun
+
+    dataset_dict = dict_from_xcom_dicts.override(task_id="make_dataset_dict")(
+        list_of_dicts="{{task_instance.xcom_pull(task_ids='run_iteration.mdrun_input_list')}}",
+        dict_structure={
+            "simulation_id": "simulation_id",
+            "tpr": "-s",
+            "xtc": "-x",
+            "gro": "-c",
+        },
+    )
+    mdrun_input_list >> dataset_dict >> mdrun
+
+
+with DAG(
+    "gmx_triggered",
+    start_date=timezone.utcnow(),
+    render_template_as_native_obj=True,
+    is_paused_upon_creation=False,
+    params={
+        "output_dir": "outputs",
+        "output_name": "sim",
+        "inputs": {
+            "mdp": {"directory": "mdp", "filename": "basic_md.json"},
+            "gro": {"directory": "ensemble_md", "filename": "sys.gro"},
+            "top": {"directory": "ensemble_md", "filename": "sys.top"},
+            "coul_lambda": 0,
+        },
+        "num_sims": 4,
+        "iteration_num": 0,
+    },
+) as callee_dag:
+    input_gro = get_file.override(task_id="get_gro")(
+        input_dir="{{ params.inputs.gro.directory }}",
+        file_name="{{ params.inputs.gro.filename }}",
+    )
+    input_top = get_file.override(task_id="get_top")(
+        input_dir="{{ params.inputs.top.directory }}",
+        file_name="{{ params.inputs.top.filename }}",
+    )
+    input_mdp = get_file.override(task_id="get_mdp")(
+        input_dir="{{ params.inputs.mdp.directory }}",
+        file_name="{{ params.inputs.mdp.filename }}",
+    )
+    mdp_sim = update_write_mdp_json_as_mdp_from_file.override(task_id="mdp_sim_update")(
+        mdp_json_file_path=input_mdp,
+        update_dict={"nsteps": 12500},
```

Comment:
It seems that coul_lambdas should also be updated. I don't see anywhere in the code really modifying lambda values in MDP files, but just having different folder names. Still this is probably fine as this example code is not for production anyway.

Reply:
All that is needed to update the coul_lambda value is to include the new value in the dictionary passed to the update_dict argument. If multiple values are needed then update_write_mdp_json_as_mdp_from_file can be called with expanded arguments, as is done in many of the example DAGs.